src/RouteCollector.php

```diff
@@ -68,17 +75,30 @@ public function route(
         array $methods = null,
         string $name = null
     ) : Route {
-        $this->checkForDuplicateRoute($path, $methods);
 
-        $methods = null === $methods ? Route::HTTP_METHOD_ANY : $methods;
+        $methods = $methods ?? Route::HTTP_METHOD_ANY;
         $route   = new Route($path, $middleware, $methods, $name);
-
+        $this->checkForDuplicateRoute($route);
+        $this->fillRouteSearchStructure($route);
         $this->routes[] = $route;
         $this->router->addRoute($route);
 
         return $route;
     }
 
+    private function fillRouteSearchStructure(Route $route): void
+    {
+        $this->routesNames[ $route->getName() ] = $route;
```

Comment:
No space following an opening brace, and no space preceding a closing brace, please. (Apply this feedback throughout the patch, please, as you've introduced the spaces throughout.)